Notification Failure

```
10:46:41.500 [pool-5-thread-1] ERROR edu.kit.ipd.crowdcontrol.objectservice.notification.Notification - An exception occurred while checking or sending a notification. java.lang.IllegalStateException: Duplicate key +---------------------+---------+------------+------------+
|id_notification_Token|result_id|result_token|notification|
+---------------------+---------+------------+------------+
|              *{null}|       *1|*???        |          *9|
+---------------------+---------+------------+------------+
at java.util.stream.Collectors.lambda$throwingMerger$0(Collectors.java:133)
at java.util.HashMap.merge(HashMap.java:1245)
at java.util.stream.Collectors.lambda$toMap$58(Collectors.java:1320)
at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
at java.util.Iterator.forEachRemaining(Iterator.java:116)
at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
at edu.kit.ipd.crowdcontrol.objectservice.notification.SQLEmailNotificationPolicy.check(SQLEmailNotificationPolicy.java:80)
at edu.kit.ipd.crowdcontrol.objectservice.notification.SQLEmailNotificationPolicy.check(SQLEmailNotificationPolicy.java:27)
at edu.kit.ipd.crowdcontrol.objectservice.notification.NotificationPolicy.invoke(NotificationPolicy.java:25)
at edu.kit.ipd.crowdcontrol.objectservice.notification.Notification.run(Notification.java:98)
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
at java.lang.Thread.run(Thread.java:745)
```

Still present:

```
Mär 22 10:57:04 edoras java[24420]: Caused by: java.lang.IllegalStateException: Duplicate key +---------------------+---------+--------------------------------------------------+------------+
Mär 22 10:57:04 edoras java[24420]: |id_notification_Token|result_id|result_token                                      |notification|
Mär 22 10:57:04 edoras java[24420]: +---------------------+---------+--------------------------------------------------+------------+
Mär 22 10:57:04 edoras java[24420]: |              *{null}|      *51|*https://workercrowdcontrol.ipd.kit.edu/ui/inde...|          *5|
Mär 22 10:57:04 edoras java[24420]: +---------------------+---------+--------------------------------------------------+------------+
Mär 22 10:57:04 edoras java[24420]: at java.util.stream.Collectors.lambda$throwingMerger$0(Collectors.java:133)
Mär 22 10:57:04 edoras java[24420]: at java.util.HashMap.merge(HashMap.java:1245)
Mär 22 10:57:04 edoras java[24420]: at java.util.stream.Collectors.lambda$toMap$58(Collectors.java:1320)
Mär 22 10:57:04 edoras java[24420]: at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
Mär 22 10:57:04 edoras java[24420]: at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
Mär 22 10:57:04 edoras java[24420]: at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
Mär 22 10:57:04 edoras java[24420]: at java.util.Iterator.forEachRemaining(Iterator.java:116)
Mär 22 10:57:04 edoras java[24420]: at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
Mär 22 10:57:04 edoras java[24420]: at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
Mär 22 10:57:04 edoras java[24420]: at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
Mär 22 10:57:04 edoras java[24420]: at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
Mär 22 10:57:04 edoras java[24420]: at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
Mär 22 10:57:04 edoras java[24420]: at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
Mär 22 10:57:04 edoras java[24420]: at edu.kit.ipd.crowdcontrol.objectservice.notification.SQLEmailNotificationPolicy.check(SQLEmailNotificationPolicy.java:87)
Mär 22 10:57:04 edoras java[24420]: ... 10 more
```

--- a/src/main/java/edu/kit/ipd/crowdcontrol/objectservice/notification/SQLEmailNotificationPolicy.java
+++ b/src/main/java/edu/kit/ipd/crowdcontrol/objectservice/notification/SQLEmailNotificationPolicy.java
@@ -86,9 +86,13 @@ public class SQLEmailNotificationPolicy extends NotificationPolicy<List<String>>
                                 .filter(record -> record.getResultId() != null)
                                 .collect(Collectors.toMap(NotificationTokenRecord::getResultId, Function.identity()));
                 } catch (java.lang.IllegalStateException e) {
-                    throw new IllegalStateException("Invalid query, the id has to be unique!", e);
+                    // TODO: Send mail to admin
+                    LOGGER.warn("Invalid query for notification {}, the id has to be unique: ", notification.getId(), e.getMessage());
+                    return Collections.emptyList();
                 } catch (DataAccessException dae) {
+                    // TODO: Send mail to admin
                     LOGGER.warn("The query for notification {} failed.", notification.getId(), dae);
+                    return Collections.emptyList();
                 }
                 List<String> newTokenList = operations.diffTokenRecords(tokenRecords, notification.getId()).stream()
                         .map(NotificationTokenRecord::getResultToken)

